### ISSUE TITLE
[Feature Request] Allow configuration of filename patterns besides `*.stories.svelte`

### ISSUE THREAD

**Issue by Bluzzi** [storybookjs/addon-svelte-csf#98]
### Describe the bug
I would like to change the name of the files in my stories, but this does not work with the addon `@storybook/addon-svelte-csf`.

### Steps to reproduce the behavior
```ts
import type { StorybookConfig } from "@storybook/sveltekit";

const config: StorybookConfig = {
  stories: ["../src/**/*.mdx", "../src/**/*.sb.svelte"], // here, I change the file extension to .sb.svelte
  addons: [
    "@storybook/addon-links",
    "@storybook/addon-essentials",
    "@storybook/addon-interactions",
    "@storybook/addon-svelte-csf"
  ],
  framework: {
    name: "@storybook/sveltekit",
    options: {},
  },
  docs: {
    autodocs: "tag",
  },
};

export default config;
```

I then create a file like `button.sb.svelte`, but it doesn't work, when I start Storybook, I get this error:
![image](https://user-images.githubusercontent.com/20624478/230074216-73a922e4-c2df-4a10-b8e0-746ae97b1adc.png)

Everything works fine if I leave the default extension (`.stories.svelte`) in the configuration and as the file name.

### Environment
 - OS: Windows
 - Node.js version: v18.15.0
 - NPM version: 8.1.1 (PNPM)

**Comment by JReinhold** [comment]
Hi!

This makes sense, the addon is only looking for `*.stories.svelte` files here, but it should probably look for `*.svelte` instead.

https://github.com/storybookjs/addon-svelte-csf/blob/main/src/preset/index.js#L56

PRs welcome!

**Comment by benmccann** [comment]
Here's the current file location: https://github.com/storybookjs/addon-svelte-csf/blob/main/src/preset/index.ts

Looking for all `*.svelte` files might be too aggressive as many won't be stories? I'd think you'd need to communicate to the plugin that it needs to look for the custom `.sb.svelte` extension

**Comment by JReinhold** [comment]
> Looking for all `*.svelte` files might be too aggressive as many won't be stories? I'd think you'd need to communicate to the plugin that it needs to look for the custom `.sb.svelte` extension

This is actually not as bad as it sounds, as the files are already pre-filtered by the `stories` configuration.
The flow is something like this:

1. Find all files that matches any of the globs in the users `stories` configuration, eg. `"../src/**/*.sb.svelte"`
2. For each file, find the first indexer that can handle this file, but testing if it's regex matches. ([The link you gave is this addon's indexer](https://github.com/storybookjs/addon-svelte-csf/blob/main/src/preset/index.ts#L62), the plain CSF indexer is another one)
3. run the file through the given indexer.

So just because the regex matches `*.svelte` doesn't mean it'll touch them. However _if_ a user configures a `stories` glob that matches `"../src/**/*.svelte"`, then that would definitely break because the indexer would try to parse non-story files.

I do agree though that a better but slightly more complex solution would be able to customize the regex via an addon option.

**Comment by benmccann** [comment]
Ah, makes sense! Probably more user friendly to avoid the extra option in that case

**Comment by shilman** [comment]
:rocket: Issue was released in [`v4.0.6`](https://github.com/storybookjs/addon-svelte-csf/releases/tag/v4.0.6) :rocket:
